File: Leetcode 75 (New)/238 Product of Array Except Self/238_ProductOfArrayExceptSelf.py
def productExceptSelf(self, nums: List[int]) -> List[int]:
    N = len(nums)
    result = [1] * N

    factor = 1

    for i in range(0, N):
        result[i] *= factor

        factor *= nums[i]

    # print(result)

    factor = 1

    for i in range(N - 1, -1, -1):
        result[i] *= factor

        factor *= nums[i]

    # print(result)

    return result

File: Leetcode 75 (New)/238 Product of Array Except Self/238_ProductOfArrayExceptSelf.java
class Solution {
    public int[] productExceptSelf(int[] nums) {

        int L = nums.length;

        int[] result = new int[L];

        int factor = 1;

        // Going forward
        for (int i = 0; i < L; i++) {

            result[i] = factor;

            factor *= nums[i];
        }

        // Going backwards
        factor = 1;

        for (int i = L - 1; i > -1; i--) {
            result[i] *= factor;

            factor *= nums[i];
        }

        return result;

    }
}